lib/improver/tests/utilities/test_spatial.py

```diff
@@ -151,82 +153,114 @@ def setUp(self):
 
     def test_basic_distance_to_grid_cells(self):
-        """Test the distance in metres to grid cell conversion."""
+        """Test the distance in metres to grid cell conversion along the
+        x-axis (default)."""
+        result = convert_distance_into_number_of_grid_cells(
+            self.cube, self.DISTANCE)
+        self.assertEqual(result, 3)
+
+    def test_distance_to_grid_cells_other_axis(self):
+        """Test the distance in metres to grid cell conversion along the
+        y-axis."""
+        self.cube.coord(axis='y').points = 0.5*self.cube.coord(axis='y').points
         result = convert_distance_into_number_of_grid_cells(
-            self.cube, self.DISTANCE,
-            max_distance_in_grid_cells=self.MAX_DISTANCE_IN_GRID_CELLS)
-        self.assertEqual(result, (3, 3))
+            self.cube, self.DISTANCE, axis='y')
+        self.assertEqual(result, 6)
 
     def test_basic_distance_to_grid_cells_float(self):
         """Test the distance in metres to grid cell conversion."""
         result = convert_distance_into_number_of_grid_cells(
-            self.cube, self.DISTANCE,
-            max_distance_in_grid_cells=self.MAX_DISTANCE_IN_GRID_CELLS,
-            int_grid_cells=False)
-        self.assertEqual(result, (3.05, 3.05))
-
-    def test_basic_no_limit(self):
-        """Test the distance in metres to grid cell conversion still works when
-        the maximum distance limit is not explicitly set."""
-        result = convert_distance_into_number_of_grid_cells(
-            self.cube, self.DISTANCE)
-        self.assertEqual(result, (3, 3))
+            self.cube, self.DISTANCE, int_grid_cells=False)
+        self.assertEqual(result, 3.05)
 
-    def test_basic_distance_to_grid_cells_different_max_distance(self):
+    def test_max_distance(self):
         """
-        Test the distance in metres to grid cell conversion for an alternative
-        max distance in grid cells.
+        Test the distance in metres to grid cell conversion within a maximum
+        distance in grid cells.
         """
-        max_distance_in_grid_cells = 50
         result = convert_distance_into_number_of_grid_cells(
-            self.cube, self.DISTANCE,
-            max_distance_in_grid_cells=max_distance_in_grid_cells)
-        self.assertEqual(result, (3, 3))
+            self.cube, self.DISTANCE, max_distance_in_grid_cells=50)
+        self.assertEqual(result, 3)
 
     def test_basic_distance_to_grid_cells_km_grid(self):
         """Test the distance-to-grid-cell conversion, grid in km."""
         self.cube.coord("projection_x_coordinate").convert_units("kilometres")
         self.cube.coord("projection_y_coordinate").convert_units("kilometres")
         result = convert_distance_into_number_of_grid_cells(
             self.cube, self.DISTANCE)
-        self.assertEqual(result, (3, 3))
+        self.assertEqual(result, 3)
 
-    def test_single_point_lat_long(self):
+    def test_error_lat_lon_grid(self):
```

Comment:
This test is a repeat and is really testing calculate_grid_spacing. It can be removed.